### ISSUE TITLE
ViewChild only works on init

### ISSUE THREAD

**Issue by mebibou** [vsternbach/angular-ts-decorators#62]
In Angular, when using `ViewChild` or `ViewChildren`, the reference of the element will be available even if the element is rendered after the view is init, so if you have a template like this:
```
<parent>
  <child *ngIf="visible"></child>
</parent>

export class Parent {
  @ViewChild(Child) child: Child;

  ngAfterViewInit() {
    setTimeout(() => visible = true, 5000);
  }
}
```
The `child` should not be null anymore after it is being rendered in the parent component. But since here the `ViewChild` is only searched in the `$postLink`, if it is not available then, it's impossible to use it

**Comment by vsternbach** [comment]
You are right, probably need to re-set it in onChanges hook, but you code still won't work in angularjs, you need to use $timeout or trigger the digest cycle in order for it to wok

**Comment by mebibou** [comment]
yes in angularjs I would use `$timeout`. Do you think it would be expensive to re-bind the children in `onChanges`?

**Comment by vsternbach** [comment]
resolved in 3e812054324edf77265a95f0b86d5af7e8f52fe0
@mebibou btw, you need to write fixes #63 in your commit in order to close the issue automatically upon merge to master

**Comment by mebibou** [comment]
ah I wrote `fix` instead of `fixes`, my bad
